micro-rdk/src/esp32/nvs_storage.rs

```diff
@@ -231,8 +231,15 @@ impl RobotConfigurationStorage for NVSStorage {
 
     fn store_robot_credentials(&self, cfg: &CloudConfig) -> Result<(), Self::Error> {
         self.set_string(NVS_ROBOT_SECRET_KEY, &cfg.secret)?;
-        self.set_string(NVS_ROBOT_ID_KEY, &cfg.id)?;
-        self.set_string(NVS_ROBOT_APP_ADDRESS, &cfg.app_address)?;
+        self.set_string(NVS_ROBOT_ID_KEY, &cfg.id).or_else(|err| {
+            let _ = self.reset_robot_credentials();
```

Comment:
- By using `reset_robot_credentials`, I think there is now a requirement that it erase keys in the same order that `store_robot_credentials` writes them, since a failed `erase_key` ends the method. Maybe it is better to explicitly just unset the keys as acted on so far in the `store` methods rather than relying on ordering in a different method? Accumulate some guards?
- ⬇️  - Is it a bug that `reset_robot_credentials` doesn't clear `NVS_ROBOT_APP_ADDRESS`?